### ISSUE TITLE
False positive when using nested list for numpy ArrayLike

### ISSUE THREAD

**Issue by matangover** [microsoft/pyright#4000]
**Describe the bug**
An error is reported for the following code even though it should not be.

**To Reproduce**
```python
import numpy as np
np.count_nonzero([[1,2,3],[4,5,6]])  # false error is reported

from numpy.typing import ArrayLike
a: ArrayLike = [[1,2,3],[4,5,6]]     # false error is reported
np.count_nonzero(a)
```

**VS Code extension or command-line**
Pylance v2022.9.40

**Additional context**
Mypy doesn't report any errors on this code.

**Comment by matangover** [comment]
Thanks for looking into it

**Comment by erictraut** [comment]
Pyright has logic to detect excessive recursion during type evaluation. This example was hitting the recursion limit, and the type evaluator was giving up. I've increased the recursion limit to allow this and similar cases to pass. There's a risk this increase could degrade performance or result in crashes (due to stack overflows), so we'll need to watch our telemetry stats carefully and roll back this change if it causes problems.

**Comment by matangover** [comment]
Thank you!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.281, which I just published. It will also be included in a future release of pylance.
